HIP/hip-415.md

```diff
@@ -0,0 +1,255 @@
+
+After the migration, Mirror nodes must be able to answer queries of type:
+- Getting block by hash/number
+- Getting a list of transactions for a block by number/hash
+- Getting logs based on block number filtering
+
+### Block Properties
+
+The following table specifies all `block` properties and at which point they will be computed. Mirror nodes must `hex` encode all properties prior to exposing them through their APIs. This table defines the properties that must be returned through APIs from Mirror Nodes.
+
+| Property | Computed By | Description  |
+| --- | --- | --- |
+| number | Services | Stored in services state (only the current number) and exported in the record stream. It is the consecutive number of the record file that is being incremented by 1 for every new record file. The number will be initially set in services through the bootstrapping process. It is exposed to the 1) EVM during Transaction Execution through the NUMBER opcode; 2) as a new property in the record file and ingested by mirror nodes. |
+| timestamp | Services | Stored in services state (only the last block timestamp) and computed by services. It is the consensusTimestamp of the first transaction in the record file. It is exposed to the 1) EVM during Transaction Execution through the TIMESTAMP opcode; 2) implicitly exported in the record file through the TS of the first transaction in the record file. |
+| hash | Services | Stored in services (last 256 blocks). It is the 32 byte prefix of the runningHash of the previous record file. That is the running hash of the last record stream object from the previous record file. It is exposed to the 1) EVM during Transaction Execution through the BLOCKHASH opcode; 2) In the record file as the End Object Running Hash |
+| baseFeePerGas | Mirror Node | Always zero, since there is no EIP-1559 style floating block capacity fees in Hedera. |
+  | difficulty | Mirror Node | Hardcoded by Mirror Node(s) to hex encoded 0 |
+  | extraData | Mirror Node | Hardcoded by Mirror Nodes(s) to 0x |
+  | gasLimit | Mirror Node |  Computed by Mirror Node(s). The gas throttle limit per second multiplied by the target block time. |
+  | gasUsed | Mirror Node | Computed by Mirror Node(s). The sum of the gasUsed value for all ContractCall and ContractCreate transactions within the block. |
+  | logsBloom | Mirror Node | Computed by Mirror Node(s). It is the bloom filter for the logs within the block. |
+  | miner | Mirror Node | Hardcoded by the Mirror Nodes to the 0.0.98 address in the long zero prefix format.  |
+  | mixHash | Mirror Node | Hardcoded by Mirror Node(s) to 0x |
+  | nonce | Mirror Node | Hardcoded by Mirror Node(s) to hex-encoded 0 |
+  | parentHash | Mirror Node | The hash of the previous block. |
+  | receiptsRoot | N/A | Hardcoded by Mirror Node(s) to hex-encoded 0. It is expected that in the future it will be 
+populated once traceability is implemented. |
```

Comment:
```suggestion
  | receiptsRoot | N/A | Hardcoded by Mirror Node(s) to hex-encoded 0. |
```
I don't expect we will ever populate it. This is core to ethereum's validation loop and since we don't validate in the same way we shouldn't imply it's any reasonable value.  This differes from uncles root, where there are never uncles so an empty list is correct.